TreeMaker/python/makeTreeFromMiniAOD_cff.py

```diff
@@ -97,9 +97,12 @@ def makeTreeFromMiniAOD(self,process):
     ## PDF weights for PDF systematics
     ## ----------------------------------------------------------------------------------------------
     if self.geninfo and self.doPDFs:
-        process.PDFWeights = cms.EDProducer('PDFWeightProducer')
-        self.VectorDouble.extend(['PDFWeights:PDFweights','PDFWeights:ScaleWeights'])
-        self.VectorInt.extend(['PDFWeights:PDFids'])
+        from TreeMaker.Utils.pdfweightproducer_cfi import PDFWeightProducer
+        process.PDFWeights = PDFWeightProducer.clone(
+            recalculatePDFs = cms.bool(self.signal),
```

Comment:
Is this what specifies that this should just be done for the SVJ signals? It seems like it will include all signals.

Reply:
The other signals should have the pdf weights already stored. It only tries to recalculate them if they aren't stored *and* if this flag is true. We can double-check this later.